src/hybrid_reader.c

```diff
@@ -117,23 +122,52 @@ static void alternatingIterate(HybridIterator *hr, VecSimQueryResult_Iterator *v
   IndexResult_Free(cur_vec_res);
 }
 
-void prepareResults(HybridIterator *hr) {
-    if (hr->mode == STANDARD_KNN) {
+void computeDistances(HybridIterator *hr) {
+  float upper_bound = INFINITY;
+  RSIndexResult *cur_res = hr->base.current;
+  RSIndexResult *cur_child_res;  // This will use the memory of hr->child->current.
+  RSIndexResult *cur_vec_res = NewDistanceResult();
+
+  while (hr->child->Read(hr->child->ctx, &cur_child_res) != INDEXREAD_EOF) {
+    float dist = (float)VecSimIndex_GetDistanceFrom(hr->index, cur_child_res->docId, hr->query.vector);
+    // If this id is not in the vector index (since it was deleted), dist will return as NaN.
+    if (isnanf(dist)) {
```

Comment:
do we have such cases? can you refer to a test?

Reply:
yes, `test_hybrid_query_non_vector_score` for example. when we update some id, the doc id remains in the index (until garbage collector is removing it) but not in the vector index

Reply:
Why change to float?